workspaces/arborist/lib/node.js

```diff
@@ -1344,9 +1344,100 @@ class Node {
     this.edgesOut.set(edge.name, edge)
   }
 
-  addEdgeIn (edge) {
+  recalculateOutEdgesOverrides () {
+    // For each edge out propogate the new overrides through.
+    for (const [, edge] of this.edgesOut) {
+      edge.reload(true)
+      if (edge.to) {
+        edge.to.updateNodeOverrideSet(edge.overrides)
+      }
+    }
+  }
+
+  findSpecificOverrideSet (first, second) {
+    for (let overrideSet = second; overrideSet; overrideSet = overrideSet.parent) {
+      if (overrideSet.isEqual(first)) {
+        return second
+      }
+    }
+    for (let overrideSet = first; overrideSet; overrideSet = overrideSet.parent) {
+      if (overrideSet.isEqual(second)) {
+        return first
+      }
+    }
+    console.log('Conflicting override sets')
```

Comment:
@wraithgar This case means that the same node has two edges with completely different override sets pointing to it. This log is useful for debugging, not sure if just to remove the line or there's some other convention used in the arborist.

Reply:
[proc-log](https://github.com/npm/cli/blob/ecd2d23d429b2fee833e534e679cce97e4190b1b/workspaces/arborist/lib/arborist/reify.js#L10) is what we use to log in npm.

[log.silly](https://github.com/npm/cli/blob/latest/workspaces/arborist/lib/arborist/reify.js#L489) is probably what we want here, defined as

> Extremely noisy excessive logging messages that are typically only useful for debugging.